sim/warlock/drain_soul.go

```diff
@@ -100,8 +98,7 @@ func (warlock *Warlock) registerDrainSoulSpell() {
 	warlock.RegisterResetEffect(func(sim *core.Simulation) {
 		sim.RegisterExecutePhaseCallback(func(sim *core.Simulation, isExecute int32) {
 			if isExecute == 25 {
-				mult := (4.0 + 0.04*float64(warlock.Talents.DeathsEmbrace)) / (1 + 0.04*float64(warlock.Talents.DeathsEmbrace))
-				warlock.DrainSoul.DamageMultiplier = mult
+				warlock.DrainSoul.DamageMultiplier *= 2.0
 			}
 		})
```

Comment:
You need to make sure you revert this execute multiplier at the end of the iteration. I think we had a way to also register cleanup actions that can be run at end of iteration so you can do it in there

Reply:
Or another and i think better option is to create an aura that modifies the damage multi on gain/expire and apply that aura when you enter execute phase. It would be automatically deactivated at the end of every iteration